src/XMakeBuildEngine/UnitTests/BackEnd/TargetUpToDateChecker_Tests.cs

```diff
@@ -903,6 +909,122 @@ bool includeOutput2
                 if (File.Exists(output2)) File.Delete(output2);
             }
         }
+
+        private static readonly DateTime Old = new DateTime(2000, 1, 1);
+        private static readonly DateTime Middle = new DateTime(2001, 1, 1);
+        private static readonly DateTime New = new DateTime(2002, 1, 1);
+
+        [Fact(Skip = "Creating a symlink on Windows requires elevation.")]
+        public void NewSymlinkOldDestinationIsUpToDate()
+        {
+            SimpleSymlinkInputCheck(symlinkWriteTime: New,
+                targetWriteTime: Old, 
+                outputWriteTime: Middle,
+                expectedOutOfDate: false);
+        }
+
+        [Fact(Skip = "Creating a symlink on Windows requires elevation.")]
+        public void OldSymlinkOldDestinationIsUpToDate()
+        {
+            SimpleSymlinkInputCheck(symlinkWriteTime: Old,
+                targetWriteTime: Middle,
+                outputWriteTime: New,
+                expectedOutOfDate: false);
+        }
+
+        [Fact(Skip = "Creating a symlink on Windows requires elevation.")]
+        public void OldSymlinkNewDestinationIsNotUpToDate()
+        {
+            SimpleSymlinkInputCheck(symlinkWriteTime: Old,
+                targetWriteTime: New,
+                outputWriteTime: Middle,
+                expectedOutOfDate: true);
+        }
+
+        [Fact(Skip = "Creating a symlink on Windows requires elevation.")]
+        public void NewSymlinkNewDestinationIsNotUpToDate()
+        {
+            SimpleSymlinkInputCheck(symlinkWriteTime: Middle,
+                targetWriteTime: Middle,
+                outputWriteTime: Old,
+                expectedOutOfDate: true);
+        }
+
+        [DllImport("kernel32.dll")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        static extern bool CreateSymbolicLink(string lpSymlinkFileName, string lpTargetFileName, UInt32 dwFlags);
+
+        [DllImport("kernel32.dll", SetLastError = true)]
+        private static extern bool SetFileTime(SafeFileHandle hFile, ref long creationTime,
+            ref long lastAccessTime, ref long lastWriteTime);
+
+        private void SimpleSymlinkInputCheck(DateTime symlinkWriteTime, DateTime targetWriteTime,
```

Comment:
I'd rename these to inputWriteTime and inputSymlinkWriteTime. At first the parameter names got me really confused.